### PR TITLE
Added an error that will be thrown if job post to be cloned doesnt exist on Canonical board

### PR DESCRIPTION
**Pull request by ilayda21**
## Done
- An error is added that will be thrown if job post to be cloned doesnt exist on Canonical board

## QA
- Checkout the feature branch
- Install dependencies with `yarn`
- Update "Job.ts" > "clonePost" method > boardPost variable :
`const boardToPost = boards.find((board) => board.name === "Test Board");`
- Run `ts-node ./src/index.ts add-posts 1753300 -c 3947968 -r emea` => [3947968](https://canonical.greenhouse.io/jobapps/3947968/edit) is not in the Canonical board.
-Verify "Job post ID: <post-id> not found in the Canonical Board." error is displayed.
- Run `ts-node ./src/index.ts add-posts 1753300 -c 3958761 -r emea` => [3958761](https://canonical.greenhouse.io/jobapps/3958761/edit) is in the Canonical board.
- Verify job posts are created with entered values.

NOTE: 
- 1753300 is [Demo Job - Scenario A](https://canonical.greenhouse.io/sdash/1753300)
- QA should be done after [this PR](https://github.com/canonical/greenhouse-automations/pull/41/files) is merged because of a bug.

## Issue
Fixes  #387

### CODE REVIEW

**Review by tbille — APPROVED** [review]
QA+10000000000000

File: src/automations/Job.ts [hunk 1]
```diff
@@ -33,22 +35,25 @@ export default class Job {
...
             // If a source post is not provided, use posts that are in the "Canonical" board.
             protectedPosts = posts.filter(
-                (post) => post.boardInfo.name === PROTECTED_JOB_BOARDS[0]
+                (post) => post.boardInfo.name === board
             );
         }
 
-        if (!protectedPosts || !protectedPosts.length)
-            throw new Error(`No post found to clone`);
+        if (!protectedPosts || !protectedPosts.length) {
```
Comment (goulinkh) [inline src/automations/Job.ts:53]: You can do something cool using the optional chaining here:
```js
if (!protectedPosts?.length)
```

File: src/automations/Job.ts [hunk 2]
```diff
@@ -10,17 +10,19 @@ import { getIDFromURL, getInnerText, joinURL } from "../common/pageUtils";
...
 import { JobInfo, PostInfo } from "../common/types";
 import Puppeteer from "puppeteer";
-import ora from "ora";
+import { Ora } from "ora";
 
 export default class Job {
     private page: Puppeteer.Page;
     private jobPost: JobPost;
     private board: Board;
+    private spinner: Ora;
```
Comment (goulinkh) [inline src/automations/Job.ts:19]: good job on the POO here!

**Review by goulinkh — APPROVED** [review]
LGTM

File: src/automations/Job.ts (hunk 1 above)
Comment (ilayda21) [reply src/automations/Job.ts]: Done :+1:

**Pull request merged**